### PR TITLE
Refactor player client

### PR DESCRIPTION
**Pull request by leungleoqin**

### CODE REVIEW

File: SOCKET-EVENTS.md
Comment (lamdbui) [inline SOCKET-EVENTS.md:1]: We may still want to document these socket events somewhere as they are still considered an interface

**Review by lamdbui — APPROVED** [review]
Just had that single comment about documenting the socket events

**Comment by lamdbui** [comment]
I'll let @Lynne-Daniels look this over as well before we do the final merge.

**Review by Lynne-Daniels — COMMENTED** [review]
good

**Pull request merged**
